library/src-bootstrapped/scala/tasty/reflect/TreeUtils.scala

```diff
@@ -287,15 +287,15 @@ trait TreeUtils
   def let(rhs: Term)(body: Ident => Term): Term = {
     import scala.quoted.QuoteContext
     given as QuoteContext = new QuoteContext(this)
-    type T // TODO probably it is better to use the Sealed contruct rather than let the user create their own existential type
-    implicit val rhsTpe: quoted.Type[T] = rhs.tpe.seal.asInstanceOf[quoted.Type[T]]
-    val rhsExpr = rhs.seal.cast[T]
-    val expr = '{
-      val x = $rhsExpr
-      ${
-        val id = ('x).unseal.asInstanceOf[Ident]
-        body(id).seal
-      }
+    val expr = (rhs.seal: @unchecked) match {
+      case '{ $rhsExpr: $t } =>
+        '{
+          val x = $rhsExpr
+          ${
+            val id = ('x).unseal.asInstanceOf[Ident]
+            body(id).seal
+          }
+        }
```

Comment:
The patmat looks really neat 👍 

However, on the other hand, as a programmer I'm a little confused: does the pattern `'{ $rhsExpr: $t }` meant to match an expression like `expr : T`?

Reply:
Kind of, but there the `T` is an existential type that will always match and give you back the type.

Reply:
We really need to write some docs for quote pattern matching.

Reply:
Is it possible to get it to work without using quote patterns but exactors? With an informative name for the extractor, it could clear the confusions.

Reply:
It might be possible at the cost of verbosity and letting another extra API for this purpose.